### PR TITLE
Apply mask to rela->r_addend to get the patching scheme

### PR DESCRIPTION
**Pull request by dezhiAmd**
#### Problem solved by the commit
Current code base uses all 32 bit of the "rela->r_addend" as patching scheme. However, according to spec, only lower 4 bit means patching scheme
#### How problem was solved, alternative solutions (if any) and why they were rejected
Apply mask to "rela->r_addend", then store it as patching scheme

### CODE REVIEW

**Review by larry9523 — APPROVED** [review]
SNL test passed

File: src/runtime_src/core/common/api/xrt_module.cpp [hunk 1]
```diff
@@ -408,6 +408,11 @@ class module_impl
 // construct patcher objects for each argument.
 class module_elf : public module_impl
 {
+  // rela->addend have offset to base-bo-addr info along with schema
+  // [0:3] bit are used for patching schema, [4:31] used for base-bo-addr
+  constexpr static uint32_t Addend_Shift = 4;
+  constexpr static uint32_t Addend_Mask = ~((uint32_t)0) << Addend_Shift;
+  constexpr static uint32_t Schema_Mask = ~Addend_Mask;
```
Comment (stsoe) [inline src/runtime_src/core/common/api/xrt_module.cpp:415]: The coding style for better or worse in `core/common` is no `Camel_Case`, so could you change to `no_camel_case`  :-)

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/common/api/xrt_module.cpp (hunk 1 above)
Comment (dezhiAmd) [reply src/runtime_src/core/common/api/xrt_module.cpp:415]: Please refer to line 678: const ELFIO::Elf32_Rela
The code style is already there.
Please...

Comment (dezhiAmd) [reply src/runtime_src/core/common/api/xrt_module.cpp:415]: There are more:

      auto begin = reinterpret_cast<const ELFIO::Elf32_Rela*>(sec->get_data());
      auto end = begin + sec->get_size() / sizeof(const ELFIO::**Elf32_Rela**);
      for (auto rela = begin; rela != end; ++rela) {
        auto symidx = ELFIO::get_sym_and_type<ELFIO::Elf32_Rela>::get_r_sym(rela->r_info);

        auto dynsym_offset = symidx * sizeof(ELFIO::**Elf32_Sym**);

Comment (stsoe) [reply src/runtime_src/core/common/api/xrt_module.cpp:415]: We can hardly do anything about code that is not our code.
Please just follow the existing coding style of the code you edit.  And it seems like the 3 lines I pointed to are indeed our code lines.

**Comment by stsoe** [comment]
When you get a chance. please run clang-tidy on xrt_module.cpp.   There are many errors we need to address.

**Pull request merged**